### PR TITLE
cellCamera: Implement cellCamera*NotifyEventQueue

### PR DESCRIPTION
**Pull request by VelocityRa**
Really old deprecated functions, used by at least 3 games from 2007:
* **Eye of Judgement** (first released game using the PS Eye)
* **EyeCreate**
* **Tori-Emaki**

Also progresses this game apparently:
* **Datura**

Includes a bit of refactoring and minor fixes.

### CODE REVIEW

File: rpcs3/Emu/Cell/Modules/cellCamera.cpp [hunk 1]
```diff
@@ -405,7 +462,7 @@ s32 cellCameraIsOpen(s32 dev_num)
 
 	const auto g_camera = fxm::get<camera_thread>();
 
-	bool is_open = g_camera->is_open;
+	const bool is_open = g_camera->is_open;
 	return is_open;
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/cellCamera.cpp]: why not just return g_camera->is_open;

File: rpcs3/Emu/Cell/Modules/cellCamera.cpp [hunk 2]
```diff
@@ -420,7 +477,7 @@ s32 cellCameraIsStarted(s32 dev_num)
 
 	const auto g_camera = fxm::get<camera_thread>();
 
-	bool is_streaming = g_camera->is_streaming;
+	const bool is_streaming = g_camera->is_streaming;
 	return is_streaming;
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/cellCamera.cpp]: why not just return g_camera->is_streaming;

File: rpcs3/Emu/Cell/Modules/cellCamera.cpp [hunk 3]
```diff
@@ -441,7 +498,7 @@ s32 cellCameraGetAttribute(s32 dev_num, s32 attrib, vm::ptr<u32> arg1, vm::ptr<u
 		return CELL_CAMERA_ERROR_PARAM;
 	}
 
-	cellCamera.todo("cellCameraGetAttribute: get attrib %s arg1: %d arg2: %d", attr_name, arg1, arg2);
+	cellCamera.todo("cellCameraGetAttribute: get attrib %s arg1: %d arg2: %d)", attr_name, arg1, arg2);
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/cellCamera.cpp]: I don't see an opening (

File: rpcs3/Emu/Cell/Modules/cellCamera.cpp [hunk 4]
```diff
@@ -623,15 +683,29 @@ s32 cellCameraReadEx(s32 dev_num, vm::ptr<CellCameraReadEx> read)
...
+
+	const auto g_camera = fxm::get<camera_thread>();
+
+	if (!g_camera)
+	{
+		return CELL_CAMERA_ERROR_NOT_INIT;
+	}
+
+	const auto status = cellCameraSetAttribute(dev_num, CELL_CAMERA_READFINISH, bufnum, arg2);
+	return status;
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/cellCamera.cpp]: why not return cellCameraSetAttribute(dev_num, CELL_CAMERA_READFINISH, bufnum, arg2);

Comment (VelocityRa) [reply rpcs3/Emu/Cell/Modules/cellCamera.cpp]: Simple, I didn't like how `return <huge func call>` looked.

File: rpcs3/Emu/Cell/Modules/cellCamera.cpp (hunk 1 above)
Comment (VelocityRa) [reply rpcs3/Emu/Cell/Modules/cellCamera.cpp]: I think these are from when I had the lock scopes and they stuck around after neko told me to remove the scopes.

File: rpcs3/Emu/Cell/Modules/cellCamera.cpp (hunk 4 above)
Comment (hcorion) [reply rpcs3/Emu/Cell/Modules/cellCamera.cpp]: It just looks funny to me, I think doing it the way MegaMouse has suggested looks best.

Comment (VelocityRa) [reply rpcs3/Emu/Cell/Modules/cellCamera.cpp]: Fair enough, I'll change it.

**Comment by hcorion** [comment]
Are there any more events that should be covered, like `CELL_CAMERA_FRAME_UPDATE`? Or will that relegated to a future PR?

**Comment by VelocityRa** [comment]
No other events needed, this particular function only registers event queues for `ATTACH` and `DETACH` notifications.

**Comment by AniLeo** [comment]
Now has conflicts after merging #4251

**Pull request merged**